db/view/view.cc

```diff
@@ -1301,16 +1303,19 @@ future<> mutate_MV(
                 return maybe_account_failure(std::move(f), std::move(*target_endpoint), false, updates_pushed_remote);
             });
             if (wait_for_all) {
-                fs->push_back(std::move(view_update));
+                fs.push_back(std::move(view_update));
             } else {
                 // The update is sent to background in order to preserve availability,
                 // its parallelism is limited by view_update_concurrency_semaphore
                 (void)view_update;
             }
         }
+        if (fs.size() >= max_concurrent_updates) {
+            co_await when_all_succeed(fs.begin(), fs.end());
```

Comment:
I think we should proceed even if there is an exception here (but still report an exception), so that we do as much work as we can.